lib/facilities/local_store.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+require 'facilities/bulk_client'
+require 'facilities/point'
+
+module Facilities
+  class LocalStore
+    attr_reader :adapter
+
+    GIS_CHECK_FREQUENCY = 1.hour
+
+    def initialize(url, _type, adapter)
+      @url = url
+      @adapter = adapter
+      @mutex = Mutex.new
+      @last_check = Time.new(0).utc
+      @current_gis_update = 0
+      @client = BulkClient.new(@url)
+      @index = {}
+      @coordinates = []
+    end
+
+    def get(id)
+      check_freshness_and_update
+      @mutex.synchronize do
+        return @index[id]
+      end
+    end
+
+    def query(bbox, services = nil)
+      check_freshness_and_update
+      @mutex.synchronize do
+        result = @coordinates.select { |p| p.within?(bbox) }
+        result = result.sort_by { |p| p.dist_from_center(bbox) }
+        result = result.map(&:facility)
+        if services
+          result = result.select { |f| @adapter.services?(f, services) }
+        end
+        result
+      end
+    end
+
+    def check_freshness_and_update
```

Comment:
How often do facilities and/or locations change? Would it be worth while caching / storing these in Redis so that we don't need to make requests for them, perhaps on a 24 hours window of time?